com.civicrm.hrjobroles/tests/phpunit/HrJobRolesTestBase.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+require_once 'CiviTest/CiviUnitTestCase.php';
+
+class HrJobRolesTestBase extends CiviUnitTestCase {
+
+
+  function setUp() {
+    $this->cleanDB();
+    parent::setUp();
+    $jobContractUpgrader = CRM_Hrjobcontract_Upgrader::instance();
+    $jobContractUpgrader->install();
+  }
+
+  function tearDown() {
+    parent::tearDown();
+  }
+
+  /**
+   * Creates a new Job Contract for the given contact
+   *
+   * If a startDate is given, it will also create a JobDetails instance to save
+   * the contract's start date and end date(if given)
+   *
+   * @param $contactID
+   * @param null $startDate
+   * @param null $endDate
+   * @param array $extraParams
+   *
+   * @return \CRM_HRJob_DAO_HRJobContract|NULL
+   */
+  public function createJobContract($contactID, $startDate = null, $endDate = null, $extraParams = array()) {
+    $contract = CRM_Hrjobcontract_BAO_HRJobContract::create(['contact_id' => $contactID]);
+    if($startDate) {
+      $params = [
+        'jobcontract_id' => $contract->id,
+        'period_start_date' => CRM_Utils_Date::processDate($startDate),
+        'period_end_date' => null,
+      ];
+      if($endDate) {
+        $params['period_end_date'] = CRM_Utils_Date::processDate($endDate);
+      }
+      $params = array_merge($params, $extraParams);
+      CRM_Hrjobcontract_BAO_HRJobDetails::create($params);
+    }
+    return $contract;
+  }
+
+  /**
+   * Creates a new Job Role with specified parameters
+   *
+   * @param $params
+   *
+   * @return \CRM_Hrjobroles_BAO_HrJobRoles|NULL
+   */
+  public function createJobRole($params) {
+    return CRM_Hrjobroles_BAO_HrJobRoles::create($params);
+  }
+
+  /**
+   * Creates sample option group and values to be used in tests
+   *
+   */
+  public function createSampleOptionGroupsAndValues()  {
+    // Create required Option Groups
+    CRM_Core_DAO::executeQuery('UPDATE civicrm_option_group SET is_active = 1');
```

Comment:
Do you really need this? It will set every option group as active, not only those you need in your test. Also, since this is not wrapped in any kind of transaction and it's not rolled back in a tearDown method, it can cause problems of test isolation (the data created for one test will still exists during another test)